msticpy/datamodel/entities/__init__.py

```diff
@@ -63,3 +65,34 @@
         "geolocation": GeoLocation,
     }
 )
+
+
+def find_entity(entity):
+    """Find entity name."""
+    entity_cf = entity.casefold()
+    entity_classes = {
+        cls.__name__.casefold(): cls for cls in Entity.ENTITY_NAME_MAP.values()
+    }
+    if entity.casefold() in Entity.ENTITY_NAME_MAP.keys():
```

Comment:
as entity_cf is already entity.casefold() could you run the if  operation on entity_cf and remove the need to casefold again?